src/app/features/auth/login/login.component.ts

```diff
@@ -0,0 +1,88 @@
+import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { AbstractControl, FormGroup, NonNullableFormBuilder, ReactiveFormsModule, Validators } from "@angular/forms";
+import { Router } from "@angular/router";
+import { AuthService } from "../services/auth.service";
+import { StorageService } from "../../../core/services/storage.service";
+import { Store } from "@ngxs/store";
+import { LoginAction, LoginTokenAction } from "@src/app/state/auth.action";
+import { Token } from "@src/app/shared/models/token";
+import { take } from "rxjs";
+
+@Component({
+  selector: 'login',
+  standalone: true,
+  imports: [CommonModule, ReactiveFormsModule],
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class LoginComponent implements OnInit {
+
+  public form: FormGroup;
+  public isSubmitted = false;
```

Comment:
This is an imperative approach, which is considered bad practice in angular development. Try to achieve the functionality without this boolean field